internal/hcs/schema2/virtual_machine.go

```diff
@@ -1,36 +1,41 @@
+// Autogenerated code; DO NOT EDIT.
+
 /*
- * HCS API
+ * Schema Open API
  *
  * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
  *
- * API version: 2.1
+ * API version: 2.4
  * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
  */
 
 package hcsschema
 
+// Configuration of a virtual machine, used during its creation to set up and/or use resources.
 type VirtualMachine struct {
-
-	// StopOnReset is private in the schema. If regenerated need to put back.
-	StopOnReset bool `json:"StopOnReset,omitempty"`
-
-	Chipset *Chipset `json:"Chipset,omitempty"`
-
-	ComputeTopology *Topology `json:"ComputeTopology,omitempty"`
-
-	Devices *Devices `json:"Devices,omitempty"`
-
-	GuestState *GuestState `json:"GuestState,omitempty"`
-
-	RestoreState *RestoreState `json:"RestoreState,omitempty"`
-
+	Version *Version `json:"Version,omitempty"`
+	// When set to true, the virtual machine will treat a reset as a stop, releasing resources and cleaning up state.
+	StopOnReset     bool             `json:"StopOnReset,omitempty"`
+	Chipset         *Chipset         `json:"Chipset,omitempty"`
+	ComputeTopology *Topology        `json:"ComputeTopology,omitempty"`
+	Devices         *Devices         `json:"Devices,omitempty"`
+	GuestState      *GuestState      `json:"GuestState,omitempty"`
+	RestoreState    *RestoreState    `json:"RestoreState,omitempty"`
 	RegistryChanges *RegistryChanges `json:"RegistryChanges,omitempty"`
-
-	StorageQoS *StorageQoS `json:"StorageQoS,omitempty"`
-
+	StorageQoS      *StorageQoS      `json:"StorageQoS,omitempty"`
+	DebugOptions    *DebugOptions    `json:"DebugOptions,omitempty"`
 	GuestConnection *GuestConnection `json:"GuestConnection,omitempty"`
-
-	SecuritySettings *SecuritySettings `json:"SecuritySettings,omitempty"`
-
-	DebugOptions *DebugOptions `json:"DebugOptions,omitempty"`
+	// Reserved field for VMMS configuration schema (embedded XML).
+	LegacySettings string `json:"LegacySettings,omitempty"`
+	// Reserved field for VMMS (VM display name).
+	DisplayName string `json:"DisplayName,omitempty"`
+	// Reserved field for VMMS (VM configuration type).
+	ConfigurationType string            `json:"ConfigurationType,omitempty"`
```

Comment:
I don't think these fields are being used anywhere so we can probably remove them

Reply:
Done.